cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -1382,6 +1382,20 @@ def _check_spatial_obs(self):
         self._validate_spatial_tissue_position("array_row", 0, 77)
         self._validate_spatial_tissue_position("in_tissue", 0, 1)
 
+        self._validate_spatial_is_primary_data()
+
+    def _validate_spatial_is_primary_data(self):
+        """
+        Validate is_primary_data for spatial datasets.
+        """
+        obs = getattr_anndata(self.adata, "obs")
+        if obs is None or "is_primary_data" not in obs:
+            return
+        if self._is_single() is False and obs["is_primary_data"].any():
```

Comment:
How are "secondary errors" typically handled? For example, should this error be reported for a non-spatial dataset that has (incorrectly) specified `uns.spatial.is_single = False`?

Reply:
since we accumulate all errors by appending to "self.errors", something like that would be handled wherever the uns check is being done. In this case, we should handle that in check_spatial_uns. I'll double check if we already are and add it in if not.

Reply:
Oh wait I see what you mean--yes, we do report secondary errors like this. That way, if the problem was that they did mean for it to be spatial but accidentally used the wrong assay, we still tell them in 1 run all the other issues associated with it. Generally, we try to report every possible error in one go even if they'd all be fixed with 1 upstream correction. With some exceptions--we don't validate the X matrix, for instance, if there are issues with dependent columns before then, because its an expensive operation to run if we know there will be failures. But we do print a warning letting them know that this was the case--`WARNING: "Validation of raw layer was not performed due to current errors, try again after fixing current errors."`

Reply:
I do think its fine that we exit early w/o checking spatial values in `check_spatial_uns` if assay is not 'EFO:0010961' (Visium Spatial Gene Expression) or 'EFO:0030062' (Slide-seqV2), since curators have not requested otherwise and the errors would end up being the same for each one (i.e. `this key should not exist if its not visium` X times)

Reply:
Got it, thanks Nayib!